lib/infrostructure/device_info/device_info.dart

```diff
@@ -1 +1,71 @@
-  
\ No newline at end of file
+import 'dart:convert';
+
+import 'package:flutter/services.dart';
+import 'package:hmi_core/hmi_core_result.dart';
+import 'package:idm_client/domain/error/failure.dart';
+import 'package:idm_client/infrostructure/device_info/api.dart';
+
+///
+/// TODO: Type doc
+class DeviceInfo {
+  String? manufacturer;
+  String? vendor;
+  String? orderCode;
+  String? model;
+  String? serial;
+  String? name;
+  String? description;
+  String? width;
+  String? height;
+  String? depth;
+  String? weight;
+  Api? _remote;
+  ///
+  /// TODO: Type doc
+  DeviceInfo({
+    this.manufacturer,
+    this.vendor,
+    this.orderCode,
+    this.model,
+    this.serial,
+    this.name,
+    this.description,
+    this.width,
+    this.height,
+    this.depth,
+    this.weight,
+  });
+  ///
+  /// TODO: Type doc
+  DeviceInfo.fromApi({
+    required String address,
+  }):
+    _remote = Api(address);
+  ///
+  /// TODO: Type doc
+  Future<Result<DeviceInfo, Failure>> fetch(String id) async {
```

Comment:
Beater choice to return `Option`/`Result` instead of `List` or `Map` when some error possible.